File: examples/python/core/wait/test-multiple-waits.py
@pytest.mark.durable_execution(handler=multiple_wait.handler)
def test_multiple_waits(durable_runner):
    """Test multiple sequential waits."""
    with durable_runner:
        result = durable_runner.run(input="test", timeout=20)

    assert result.status is InvocationStatus.SUCCEEDED
    
    # Find wait operations by name
    wait_1 = result.get_wait("wait-1")
    wait_2 = result.get_wait("wait-2")
    
    assert wait_1.scheduled_end_timestamp is not None
    assert wait_2.scheduled_end_timestamp is not None

File: examples/csharp/core/wait/test-multiple-waits.cs
using System.Linq;
using Amazon.Lambda.DurableExecution;
using Amazon.Lambda.DurableExecution.Testing;
using Xunit;

public class MultipleWaitsTests
{
    [Fact]
    public async Task HandlesMultipleSequentialWaits()
    {
        // SkipTime fast-forwards waits so the test runs instantly.
        await using var runner = new DurableTestRunner<object, MultipleWaitsResult>(
            handler: Workflow,
            options: new TestRunnerOptions { SkipTime = true });

        var result = await runner.RunAsync("test");
        result.EnsureSucceeded();

        Assert.Equal(2, result.Result!.CompletedWaits);
        Assert.Equal("done", result.Result!.FinalStep);

        // Both waits are recorded with their names.
        var waitOps = result.Steps.Where(s => s.Kind == OperationKind.Wait).ToList();
        Assert.Equal(2, waitOps.Count);

        var waitNames = waitOps.Select(w => w.Name).ToList();
        Assert.Contains("wait-1", waitNames);
        Assert.Contains("wait-2", waitNames);
    }

    private static async Task<MultipleWaitsResult> Workflow(object input, IDurableContext ctx)
    {
        await ctx.WaitAsync(TimeSpan.FromMinutes(5), name: "wait-1");
        await ctx.WaitAsync(TimeSpan.FromHours(1), name: "wait-2");
        return new MultipleWaitsResult(CompletedWaits: 2, FinalStep: "done");
    }
}

public record MultipleWaitsResult(int CompletedWaits, string FinalStep);
